extensions/schema-compare/src/test/schemaCompare.test.ts

```diff
@@ -178,8 +178,70 @@ describe('SchemaCompareMainWindow.execute', function (): void {
 		await result.execute();
 
 		//Generate script button and apply button should be enabled for database comparison
-		should(result.verifyButtonsState(true, true)).equal(true);
+		should(result.verifyButtonsState(true, true, true, true, true, false, true, true, true, true)).equal(true);
 	});
 
 });
 
+describe('SchemaCompareMainWindow.updateSourceAndTarget', function (): void {
+	before(() => {
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
+		testContext = createContext();
+	});
+
+	it('Should set buttons appropriately when source and target endpoints are empty', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService();
+		let endpointInfo: mssql.SchemaCompareEndpointInfo;
+
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
```

Comment:
I think we were trying to switch over to use undefined instead of null in ADS

Reply:
Fixed it.